src/classes/RD2_VisualizeScheduleController.cls

```diff
@@ -0,0 +1,150 @@
+/*
+    Copyright (c) 2019 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Controller for the Schedule Visualization Component
+*
+*/
+public with sharing class RD2_VisualizeScheduleController {
```

Comment:
I'm considering using this same controller class for the LWC that renders the active schedule records since there are a number of parts of this that are similar. I can see a couple of approaches, but I'm curious what you think would work well here.

- Rename this class to `RD2_ScheduleController` and have methods for both LWC components. That's likely `getInstallments()` (renamed from `getSchedule`) and `getActiveSchedule()`.
- Create a base abstract class for `RD2_ScheduleController` that has the common methods and use one controller class per LWC that extends from the base: `RD2_VisualizeScheduleController` and `RD2_ShowActiveScheduleController`

Or the third option is to just have two different classes without any overlap.

Reply:
I think it might be pretty straightforward to have one controller for both components, but I guess it depends at least partially on how many additional items this might be used for.  If you think it will expand further, it might make some sense to go the abstract base route for the stuff related to RD and currency code.  Or, if it offends our standard approach to not have a one apex controller to one javascript controller model, I suppose separate would be okay too.  But my vote is for one.